src/__internal__/form-field/form-field.component.tsx

```diff
@@ -129,6 +129,11 @@ const FormField = ({
       "Use `readOnly` if you require users to see validations with a non-interactive field"
   );
 
+  invariant(
```

Comment:
praise: good idea to add a guard against setting both props 👏

Reply:
suggestion: I think the conditional check needs tweaking slightly though. This will actually only fire when neither is set, we actually want it to fire if **both** are set at the same time

```
isRequired && isOptional

Reply:
sorry it needs to be `!(isRequired && isOptional)`